### PR TITLE
[MNG-6829] Replace any StringUtils#isEmpty(String) and #isNotEmpty(String)

### PR DESCRIPTION
**Pull request by timtebeek**
### [Replace any StringUtils#isEmpty(String) and #isNotEmpty(String)](https://public.moderne.io/recipes/org.openrewrite.java.migrate.apache.commons.lang.IsNotEmptyToJdk)

Followup from #120 for @elharo

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/pmd/PmdViolationCheckMojo.java [hunk 1]
```diff
@@ -143,7 +142,7 @@ protected ViolationDetails<Violation> newViolationDetailsInstance() {
     private String getFilename(String fullpath, String pkg) {
         int index = fullpath.lastIndexOf(File.separatorChar);
 
-        while (StringUtils.isNotEmpty(pkg)) {
+        while ((pkg != null && !pkg.isEmpty())) {
```
Comment (timtebeek) [inline src/main/java/org/apache/maven/plugins/pmd/PmdViolationCheckMojo.java:145]:
```suggestion
        while (pkg != null && !pkg.isEmpty()) {
```

File: src/main/java/org/apache/maven/plugins/pmd/ExcludeViolationsFromFile.java [hunk 2]
```diff
@@ -107,9 +106,9 @@ private boolean isExcludedFromFailure(String className, String ruleName) {
     private String extractClassName(String packageName, String className, String fullPath) {
         // for some reason, some violations don't contain the package name, so we have to guess the full class name
         // this looks like a bug in PMD - at least for UnusedImport rule.
-        if (StringUtils.isNotEmpty(packageName) && StringUtils.isNotEmpty(className)) {
+        if ((packageName != null && !packageName.isEmpty()) && (className != null && !className.isEmpty())) {
```
Comment (timtebeek) [inline src/main/java/org/apache/maven/plugins/pmd/ExcludeViolationsFromFile.java:109]:
```suggestion
        if (packageName != null && !packageName.isEmpty() && className != null && !className.isEmpty()) {
```

**Review by timtebeek — COMMENTED** [review]
Minor touch up.

**Review by elharo — APPROVED** [review]

**Comment by elharo** [comment]
You can assign these reviews to me. I might not get to all of them this quickly but they're not hard to review.

I would like it if you can amend the commit messages so they no longer say "Co-authored-by: Moderne".

Longer term, we might want to look into a flow analysis tool that can tell us when the null checks are unneeded, but baby steps for now.

**Comment by elharo** [comment]
The "co-authored" message has the potential to cause licensing and ownership issues. Alternately, Moderne could assign the  corporate CLA: https://www.apache.org/licenses/cla-corporate.pdf

**Comment by elharo** [comment]
In fact, now that I think about it, we should pause these changes until Moderne does sign the corporate CLA. An Individual CLA is insufficient given the co-authorship line.

**Comment by timtebeek** [comment]
> In fact, now that I think about it, we should pause these changes until Moderne does sign the corporate CLA. An Individual CLA is insufficient given the co-authorship line.

@elharo I'm pleased to say we've finally been able to sign (and get accepted) the CCLA:

> This message acknowledges receipt of the following document, which has been filed in the Apache Software Foundation records:
>  CCLA from Moderne Inc.

Would you want me to resume creating pull requests to replace `StringUtils#is/Not/Empty(String)` as we've done here?
I was thinking to open up to ten at a time, and only start a next batch when there's less than five open; does that sound OK?

The changes would be made by running this recipe against the Apache Maven organization defined in the top right corner:
https://public.moderne.io/recipes/org.openrewrite.java.migrate.apache.commons.lang.IsNotEmptyToJdk
You're more than welcome to have a look as well; shouldn't take more than a couple minutes to create a few PRs or direct commits.

**Pull request merged**